// samp-node-lib/build/Game.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Game = void 0;
const SampEvents_1 = require("./SampEvents");
class Game {
    OnGameModeInit(func) {
        return (0, SampEvents_1.OnGameModeInit)(func);
    }
    OnGameModeExit(func) {
        return (0, SampEvents_1.OnGameModeExit)(func);
    }
    OnFilterScriptInit(func) {
        return (0, SampEvents_1.OnFilterScriptInit)(func);
    }
    OnFilterScriptExit(func) {
        return (0, SampEvents_1.OnFilterScriptExit)(func);
    }
    OnPlayerConnect(func) {
        return (0, SampEvents_1.OnPlayerConnect)(func);
    }
    OnPlayerDisconnect(func) {
        return (0, SampEvents_1.OnPlayerDisconnect)(func);
    }
    OnPlayerSpawn(func) {
        return (0, SampEvents_1.OnPlayerSpawn)(func);
    }
    OnPlayerDeath(func) {
        return (0, SampEvents_1.OnPlayerDeath)(func);
    }
    OnVehicleSpawn(func) {
        return (0, SampEvents_1.OnVehicleSpawn)(func);
    }
    OnVehicleDeath(func) {
        return (0, SampEvents_1.OnVehicleDeath)(func);
    }
    OnPlayerText(func) {
        return (0, SampEvents_1.OnPlayerText)(func);
    }
    OnPlayerCommandText(func) {
        return (0, SampEvents_1.OnPlayerCommandText)(func);
    }
    OnPlayerRequestClass(func) {
        return (0, SampEvents_1.OnPlayerRequestClass)(func);
    }
    OnPlayerEnterVehicle(func) {
        return (0, SampEvents_1.OnPlayerEnterVehicle)(func);
    }
    OnPlayerExitVehicle(func) {
        return (0, SampEvents_1.OnPlayerExitVehicle)(func);
    }
    OnPlayerStateChange(func) {
        return (0, SampEvents_1.OnPlayerStateChange)(func);
    }
    OnPlayerEnterCheckpoint(func) {
        return (0, SampEvents_1.OnPlayerEnterCheckpoint)(func);
    }
    OnPlayerLeaveCheckpoint(func) {
        return (0, SampEvents_1.OnPlayerLeaveCheckpoint)(func);
    }
    OnPlayerEnterRaceCheckpoint(func) {
        return (0, SampEvents_1.OnPlayerEnterRaceCheckpoint)(func);
    }
    OnPlayerLeaveRaceCheckpoint(func) {
        return (0, SampEvents_1.OnPlayerLeaveRaceCheckpoint)(func);
    }
    OnRconCommand(func) {
        return (0, SampEvents_1.OnRconCommand)(func);
    }
    OnPlayerRequestSpawn(func) {
        return (0, SampEvents_1.OnPlayerRequestSpawn)(func);
    }
    OnObjectMoved(func) {
        return (0, SampEvents_1.OnObjectMoved)(func);
    }
    OnPlayerObjectMoved(func) {
        return (0, SampEvents_1.OnPlayerObjectMoved)(func);
    }
    OnPlayerPickUpPickup(func) {
        return (0, SampEvents_1.OnPlayerPickUpPickup)(func);
    }
    OnVehicleMod(func) {
        return (0, SampEvents_1.OnVehicleMod)(func);
    }
    OnEnterExitModShop(func) {
        return (0, SampEvents_1.OnEnterExitModShop)(func);
    }
    OnVehiclePaintjob(func) {
        return (0, SampEvents_1.OnVehiclePaintjob)(func);
    }
    OnVehicleRespray(func) {
        return (0, SampEvents_1.OnVehicleRespray)(func);
    }
    OnVehicleDamageStatusUpdate(func) {
        return (0, SampEvents_1.OnVehicleDamageStatusUpdate)(func);
    }
    OnUnoccupiedVehicleUpdate(func) {
        return (0, SampEvents_1.OnUnoccupiedVehicleUpdate)(func);
    }
    OnPlayerSelectedMenuRow(func) {
        return (0, SampEvents_1.OnPlayerSelectedMenuRow)(func);
    }
    OnPlayerExitedMenu(func) {
        return (0, SampEvents_1.OnPlayerExitedMenu)(func);
    }
    OnPlayerInteriorChange(func) {
        return (0, SampEvents_1.OnPlayerInteriorChange)(func);
    }
    OnPlayerKeyStateChange(func) {
        return (0, SampEvents_1.OnPlayerKeyStateChange)(func);
    }
    OnRconLoginAttempt(func) {
        return (0, SampEvents_1.OnRconLoginAttempt)(func);
    }
    OnPlayerUpdate(func) {
        return (0, SampEvents_1.OnPlayerUpdate)(func);
    }
    OnPlayerStreamIn(func) {
        return (0, SampEvents_1.OnPlayerStreamIn)(func);
    }
    OnPlayerStreamOut(func) {
        return (0, SampEvents_1.OnPlayerStreamOut)(func);
    }
    OnVehicleStreamIn(func) {
        return (0, SampEvents_1.OnVehicleStreamIn)(func);
    }
    OnVehicleStreamOut(func) {
        return (0, SampEvents_1.OnVehicleStreamOut)(func);
    }
    OnActorStreamIn(func) {
        return (0, SampEvents_1.OnActorStreamIn)(func);
    }
    OnActorStreamOut(func) {
        return (0, SampEvents_1.OnActorStreamOut)(func);
    }
    OnDialogResponse(func) {
        return (0, SampEvents_1.OnDialogResponse)(func);
    }
    OnPlayerTakeDamage(func) {
        return (0, SampEvents_1.OnPlayerTakeDamage)(func);
    }
    OnPlayerGiveDamage(func) {
        return (0, SampEvents_1.OnPlayerGiveDamage)(func);
    }
    OnPlayerGiveDamageActor(func) {
        return (0, SampEvents_1.OnPlayerGiveDamageActor)(func);
    }
    OnPlayerClickMap(func) {
        return (0, SampEvents_1.OnPlayerClickMap)(func);
    }
    OnPlayerClickTextDraw(func) {
        return (0, SampEvents_1.OnPlayerClickTextDraw)(func);
    }
    OnPlayerClickPlayerTextDraw(func) {
        return (0, SampEvents_1.OnPlayerClickPlayerTextDraw)(func);
    }
    OnIncomingConnection(func) {
        return (0, SampEvents_1.OnIncomingConnection)(func);
    }
    OnTrailerUpdate(func) {
        return (0, SampEvents_1.OnTrailerUpdate)(func);
    }
    OnVehicleSirenStateChange(func) {
        return (0, SampEvents_1.OnVehicleSirenStateChange)(func);
    }
    OnPlayerFinishedDownloading(func) {
        return (0, SampEvents_1.OnPlayerFinishedDownloading)(func);
    }
    OnPlayerClickPlayer(func) {
        return (0, SampEvents_1.OnPlayerClickPlayer)(func);
    }
    OnPlayerSelectObject(func) {
        return (0, SampEvents_1.OnPlayerSelectObject)(func);
    }
    OnPlayerWeaponShot(func) {
        return (0, SampEvents_1.OnPlayerWeaponShot)(func);
    }
    OnClientCheckResponse(func) {
        return (0, SampEvents_1.OnClientCheckResponse)(func);
    }
    OnScriptCash(func) {
        return (0, SampEvents_1.OnScriptCash)(func);
    }
}
exports.Game = Game;
