// samp-node-lib/build/SampEvents.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.OnPlayerClickPlayerTextDraw = exports.OnPlayerClickTextDraw = exports.OnPlayerClickMap = exports.OnPlayerGiveDamageActor = exports.OnPlayerGiveDamage = exports.OnPlayerTakeDamage = exports.OnDialogResponse = exports.OnActorStreamOut = exports.OnActorStreamIn = exports.OnVehicleStreamOut = exports.OnVehicleStreamIn = exports.OnPlayerStreamOut = exports.OnPlayerStreamIn = exports.OnPlayerUpdate = exports.OnRconLoginAttempt = exports.OnPlayerKeyStateChange = exports.OnPlayerInteriorChange = exports.OnPlayerExitedMenu = exports.OnPlayerSelectedMenuRow = exports.OnUnoccupiedVehicleUpdate = exports.OnVehicleDamageStatusUpdate = exports.OnVehicleRespray = exports.OnVehiclePaintjob = exports.OnEnterExitModShop = exports.OnVehicleMod = exports.OnPlayerPickUpPickup = exports.OnPlayerObjectMoved = exports.OnObjectMoved = exports.OnPlayerRequestSpawn = exports.OnRconCommand = exports.OnPlayerLeaveRaceCheckpoint = exports.OnPlayerEnterRaceCheckpoint = exports.OnPlayerLeaveCheckpoint = exports.OnPlayerEnterCheckpoint = exports.OnPlayerStateChange = exports.OnPlayerExitVehicle = exports.OnPlayerEnterVehicle = exports.OnPlayerRequestClass = exports.OnPlayerCommandText = exports.OnPlayerText = exports.OnVehicleDeath = exports.OnVehicleSpawn = exports.OnPlayerDeath = exports.OnPlayerSpawn = exports.OnPlayerDisconnect = exports.OnPlayerConnect = exports.OnFilterScriptExit = exports.OnFilterScriptInit = exports.OnGameModeExit = exports.OnGameModeInit = void 0;
exports.OnScriptCash = exports.OnClientCheckResponse = exports.OnPlayerWeaponShot = exports.OnPlayerSelectObject = exports.OnPlayerClickPlayer = exports.OnPlayerFinishedDownloading = exports.OnVehicleSirenStateChange = exports.OnTrailerUpdate = exports.OnIncomingConnection = void 0;
const SampPlayers_1 = require("./SampPlayers");
const SampVehicles_1 = require("./SampVehicles");
const OnGameModeInit = (func) => {
    samp.on('OnGameModeInit', func);
};
exports.OnGameModeInit = OnGameModeInit;
const OnGameModeExit = (func) => {
    samp.on('OnGameModeExit', func);
};
exports.OnGameModeExit = OnGameModeExit;
const OnFilterScriptInit = (func) => {
    samp.on('OnFilterScriptInit', func);
};
exports.OnFilterScriptInit = OnFilterScriptInit;
const OnFilterScriptExit = (func) => {
    samp.on('OnFilterScriptExit', func);
};
exports.OnFilterScriptExit = OnFilterScriptExit;
const OnPlayerConnect = (func) => {
    samp.on('OnPlayerConnect', ((playerid) => func(SampPlayers_1.SampPlayers.getNewClass(playerid))));
};
exports.OnPlayerConnect = OnPlayerConnect;
const OnPlayerDisconnect = (func) => {
    samp.on('OnPlayerDisconnect', ((playerid, reason) => func(SampPlayers_1.SampPlayers.removeClass(playerid), reason)));
};
exports.OnPlayerDisconnect = OnPlayerDisconnect;
const OnPlayerSpawn = (func) => {
    samp.on('OnPlayerSpawn', ((playerid) => func(SampPlayers_1.SampPlayers.getClass(playerid))));
};
exports.OnPlayerSpawn = OnPlayerSpawn;
const OnPlayerDeath = (func) => {
    samp.on('OnPlayerDeath', ((playerid, killerid, reason) => func(SampPlayers_1.SampPlayers.getClass(playerid), SampPlayers_1.SampPlayers.getClass(killerid), reason)));
};
exports.OnPlayerDeath = OnPlayerDeath;
const OnVehicleSpawn = (func) => {
    samp.on('OnVehicleSpawn', (vehicleid) => func(SampVehicles_1.SampVehicles.getNewClass(vehicleid)));
};
exports.OnVehicleSpawn = OnVehicleSpawn;
const OnVehicleDeath = (func) => {
    samp.on('OnVehicleDeath', ((vehicleid, playerid) => func(SampVehicles_1.SampVehicles.removeClass(vehicleid), SampPlayers_1.SampPlayers.getClass(playerid))));
};
exports.OnVehicleDeath = OnVehicleDeath;
const OnPlayerText = (func) => {
    samp.on('OnPlayerText', ((playerid, text) => func(SampPlayers_1.SampPlayers.getClass(playerid), text)));
};
exports.OnPlayerText = OnPlayerText;
const OnPlayerCommandText = (func) => {
    samp.on('OnPlayerCommandText', ((playerid, cmdtext) => func(SampPlayers_1.SampPlayers.getClass(playerid), cmdtext)));
};
exports.OnPlayerCommandText = OnPlayerCommandText;
const OnPlayerRequestClass = (func) => {
    samp.on('OnPlayerRequestClass', ((playerid, classid) => func(SampPlayers_1.SampPlayers.getClass(playerid), classid)));
};
exports.OnPlayerRequestClass = OnPlayerRequestClass;
const OnPlayerEnterVehicle = (func) => {
    samp.on('OnPlayerEnterVehicle', ((playerid, vehicleid, ispassenger) => func(SampPlayers_1.SampPlayers.getClass(playerid), SampVehicles_1.SampVehicles.getClass(vehicleid), ispassenger)));
};
exports.OnPlayerEnterVehicle = OnPlayerEnterVehicle;
const OnPlayerExitVehicle = (func) => {
    samp.on('OnPlayerExitVehicle', ((playerid, vehicleid) => func(SampPlayers_1.SampPlayers.getClass(playerid), SampVehicles_1.SampVehicles.getClass(vehicleid))));
};
exports.OnPlayerExitVehicle = OnPlayerExitVehicle;
const OnPlayerStateChange = (func) => {
    samp.on('OnPlayerStateChange', ((playerid, newstate, oldstate) => func(SampPlayers_1.SampPlayers.getClass(playerid), newstate, oldstate)));
};
exports.OnPlayerStateChange = OnPlayerStateChange;
const OnPlayerEnterCheckpoint = (func) => {
    samp.on('OnPlayerEnterCheckpoint', ((playerid) => func(SampPlayers_1.SampPlayers.getClass(playerid))));
};
exports.OnPlayerEnterCheckpoint = OnPlayerEnterCheckpoint;
const OnPlayerLeaveCheckpoint = (func) => {
    samp.on('OnPlayerLeaveCheckpoint', ((playerid) => func(SampPlayers_1.SampPlayers.getClass(playerid))));
};
exports.OnPlayerLeaveCheckpoint = OnPlayerLeaveCheckpoint;
const OnPlayerEnterRaceCheckpoint = (func) => {
    samp.on('OnPlayerEnterRaceCheckpoint', ((playerid) => func(SampPlayers_1.SampPlayers.getClass(playerid))));
};
exports.OnPlayerEnterRaceCheckpoint = OnPlayerEnterRaceCheckpoint;
const OnPlayerLeaveRaceCheckpoint = (func) => {
    samp.on('OnPlayerLeaveRaceCheckpoint', ((playerid) => func(SampPlayers_1.SampPlayers.getClass(playerid))));
};
exports.OnPlayerLeaveRaceCheckpoint = OnPlayerLeaveRaceCheckpoint;
const OnRconCommand = (func) => {
    samp.on('OnRconCommand', func);
};
exports.OnRconCommand = OnRconCommand;
const OnPlayerRequestSpawn = (func) => {
    samp.on('OnPlayerRequestSpawn', ((playerid) => func(SampPlayers_1.SampPlayers.getClass(playerid))));
};
exports.OnPlayerRequestSpawn = OnPlayerRequestSpawn;
const OnObjectMoved = (func) => {
    samp.on('OnObjectMoved', func);
};
exports.OnObjectMoved = OnObjectMoved;
const OnPlayerObjectMoved = (func) => {
    samp.on('OnPlayerObjectMoved', ((playerid, objectid) => func(SampPlayers_1.SampPlayers.getClass(playerid), objectid)));
};
exports.OnPlayerObjectMoved = OnPlayerObjectMoved;
const OnPlayerPickUpPickup = (func) => {
    samp.on('OnPlayerPickUpPickup', ((playerid, pickupid) => func(SampPlayers_1.SampPlayers.getClass(playerid), pickupid)));
};
exports.OnPlayerPickUpPickup = OnPlayerPickUpPickup;
const OnVehicleMod = (func) => {
    samp.on('OnVehicleMod', ((playerid, vehicleid, componentid) => func(SampPlayers_1.SampPlayers.getClass(playerid), SampVehicles_1.SampVehicles.getClass(vehicleid), componentid)));
};
exports.OnVehicleMod = OnVehicleMod;
const OnEnterExitModShop = (func) => {
    samp.on('OnEnterExitModShop', ((playerid, enterexit, interiorid) => func(SampPlayers_1.SampPlayers.getClass(playerid), enterexit, interiorid)));
};
exports.OnEnterExitModShop = OnEnterExitModShop;
const OnVehiclePaintjob = (func) => {
    samp.on('OnVehiclePaintjob', ((playerid, vehicleid, paintjobid) => func(SampPlayers_1.SampPlayers.getClass(playerid), SampVehicles_1.SampVehicles.getClass(vehicleid), paintjobid)));
};
exports.OnVehiclePaintjob = OnVehiclePaintjob;
const OnVehicleRespray = (func) => {
    samp.on('OnVehicleRespray', ((playerid, vehicleid, color1, color2) => func(SampPlayers_1.SampPlayers.getClass(playerid), SampVehicles_1.SampVehicles.getClass(vehicleid), color1, color2)));
};
exports.OnVehicleRespray = OnVehicleRespray;
const OnVehicleDamageStatusUpdate = (func) => {
    samp.on('OnVehicleDamageStatusUpdate', ((vehicleid, playerid) => func(SampVehicles_1.SampVehicles.getClass(vehicleid), SampPlayers_1.SampPlayers.getClass(playerid))));
};
exports.OnVehicleDamageStatusUpdate = OnVehicleDamageStatusUpdate;
const OnUnoccupiedVehicleUpdate = (func) => {
    samp.on('OnUnoccupiedVehicleUpdate', ((vehicleid, playerid, passenger_seat, new_x, new_y, new_z, vel_x, vel_y, vel_z) => func(SampVehicles_1.SampVehicles.getClass(vehicleid), SampPlayers_1.SampPlayers.getClass(playerid), passenger_seat, new_x, new_y, new_z, vel_x, vel_y, vel_z)));
};
exports.OnUnoccupiedVehicleUpdate = OnUnoccupiedVehicleUpdate;
const OnPlayerSelectedMenuRow = (func) => {
    samp.on('OnPlayerSelectedMenuRow', ((playerid, row) => func(SampPlayers_1.SampPlayers.getClass(playerid), row)));
};
exports.OnPlayerSelectedMenuRow = OnPlayerSelectedMenuRow;
const OnPlayerExitedMenu = (func) => {
    samp.on('OnPlayerExitedMenu', ((playerid) => func(SampPlayers_1.SampPlayers.getClass(playerid))));
};
exports.OnPlayerExitedMenu = OnPlayerExitedMenu;
const OnPlayerInteriorChange = (func) => {
    samp.on('OnPlayerInteriorChange', ((playerid, newinteriorid, oldinteriorid) => func(SampPlayers_1.SampPlayers.getClass(playerid), newinteriorid, oldinteriorid)));
};
exports.OnPlayerInteriorChange = OnPlayerInteriorChange;
const OnPlayerKeyStateChange = (func) => {
    samp.on('OnPlayerKeyStateChange', ((playerid, newkeys, oldkeys) => func(SampPlayers_1.SampPlayers.getClass(playerid), newkeys, oldkeys)));
};
exports.OnPlayerKeyStateChange = OnPlayerKeyStateChange;
const OnRconLoginAttempt = (func) => {
    samp.on('OnRconLoginAttempt', func);
};
exports.OnRconLoginAttempt = OnRconLoginAttempt;
const OnPlayerUpdate = (func) => {
    samp.on('OnPlayerUpdate', ((playerid) => func(SampPlayers_1.SampPlayers.getClass(playerid))));
};
exports.OnPlayerUpdate = OnPlayerUpdate;
const OnPlayerStreamIn = (func) => {
    samp.on('OnPlayerStreamIn', ((playerid, forplayerid) => func(SampPlayers_1.SampPlayers.getClass(playerid), SampPlayers_1.SampPlayers.getClass(forplayerid))));
};
exports.OnPlayerStreamIn = OnPlayerStreamIn;
const OnPlayerStreamOut = (func) => {
    samp.on('OnPlayerStreamOut', ((playerid, forplayerid) => func(SampPlayers_1.SampPlayers.getClass(playerid), SampPlayers_1.SampPlayers.getClass(forplayerid))));
};
exports.OnPlayerStreamOut = OnPlayerStreamOut;
const OnVehicleStreamIn = (func) => {
    samp.on('OnVehicleStreamIn', ((vehicleid, forplayerid) => func(SampVehicles_1.SampVehicles.getClass(vehicleid), SampPlayers_1.SampPlayers.getClass(forplayerid))));
};
exports.OnVehicleStreamIn = OnVehicleStreamIn;
const OnVehicleStreamOut = (func) => {
    samp.on('OnVehicleStreamOut', ((vehicleid, forplayerid) => func(SampVehicles_1.SampVehicles.getClass(vehicleid), SampPlayers_1.SampPlayers.getClass(forplayerid))));
};
exports.OnVehicleStreamOut = OnVehicleStreamOut;
const OnActorStreamIn = (func) => {
    samp.on('OnActorStreamIn', ((actorid, forplayerid) => func(actorid, SampPlayers_1.SampPlayers.getClass(forplayerid))));
};
exports.OnActorStreamIn = OnActorStreamIn;
const OnActorStreamOut = (func) => {
    samp.on('OnActorStreamOut', ((actorid, forplayerid) => func(actorid, SampPlayers_1.SampPlayers.getClass(forplayerid))));
};
exports.OnActorStreamOut = OnActorStreamOut;
const OnDialogResponse = (func) => {
    samp.on('OnDialogResponse', ((playerid, dialogid, response, listitem, inputtext) => func(SampPlayers_1.SampPlayers.getClass(playerid), dialogid, response, listitem, inputtext)));
};
exports.OnDialogResponse = OnDialogResponse;
const OnPlayerTakeDamage = (func) => {
    samp.on('OnPlayerTakeDamage', ((playerid, issuerid, amount, weaponid, bodypart) => func(SampPlayers_1.SampPlayers.getClass(playerid), SampPlayers_1.SampPlayers.getClass(issuerid), amount, weaponid, bodypart)));
};
exports.OnPlayerTakeDamage = OnPlayerTakeDamage;
const OnPlayerGiveDamage = (func) => {
    samp.on('OnPlayerGiveDamage', ((playerid, damagedid, amount, weaponid, bodypart) => func(SampPlayers_1.SampPlayers.getClass(playerid), SampPlayers_1.SampPlayers.getClass(damagedid), amount, weaponid, bodypart)));
};
exports.OnPlayerGiveDamage = OnPlayerGiveDamage;
const OnPlayerGiveDamageActor = (func) => {
    samp.on('OnPlayerGiveDamageActor', ((playerid, damaged_actorid, amount, weaponid, bodypart) => func(SampPlayers_1.SampPlayers.getClass(playerid), damaged_actorid, amount, weaponid, bodypart)));
};
exports.OnPlayerGiveDamageActor = OnPlayerGiveDamageActor;
const OnPlayerClickMap = (func) => {
    samp.on('OnPlayerClickMap', ((playerid, fX, fY, fZ) => func(SampPlayers_1.SampPlayers.getClass(playerid), fX, fY, fZ)));
};
exports.OnPlayerClickMap = OnPlayerClickMap;
const OnPlayerClickTextDraw = (func) => {
    samp.on('OnPlayerClickTextDraw', ((playerid, clickedid) => func(SampPlayers_1.SampPlayers.getClass(playerid), clickedid)));
};
exports.OnPlayerClickTextDraw = OnPlayerClickTextDraw;
const OnPlayerClickPlayerTextDraw = (func) => {
    samp.on('OnPlayerClickPlayerTextDraw', ((playerid, playertextid) => func(SampPlayers_1.SampPlayers.getClass(playerid), playertextid)));
};
exports.OnPlayerClickPlayerTextDraw = OnPlayerClickPlayerTextDraw;
const OnIncomingConnection = (func) => {
    samp.on('OnIncomingConnection', ((playerid, ip_address, port) => func(SampPlayers_1.SampPlayers.getClass(playerid), ip_address, port)));
};
exports.OnIncomingConnection = OnIncomingConnection;
const OnTrailerUpdate = (func) => {
    samp.on('OnTrailerUpdate', ((playerid, vehicleid) => func(SampPlayers_1.SampPlayers.getClass(playerid), SampVehicles_1.SampVehicles.getClass(vehicleid))));
};
exports.OnTrailerUpdate = OnTrailerUpdate;
const OnVehicleSirenStateChange = (func) => {
    samp.on('OnVehicleSirenStateChange', ((playerid, vehicleid, newstate) => func(SampPlayers_1.SampPlayers.getClass(playerid), SampVehicles_1.SampVehicles.getClass(vehicleid), newstate)));
};
exports.OnVehicleSirenStateChange = OnVehicleSirenStateChange;
const OnPlayerFinishedDownloading = (func) => {
    samp.on('OnPlayerFinishedDownloading', ((playerid, virtualworld) => func(SampPlayers_1.SampPlayers.getClass(playerid), virtualworld)));
};
exports.OnPlayerFinishedDownloading = OnPlayerFinishedDownloading;
const OnPlayerClickPlayer = (func) => {
    samp.on('OnPlayerClickPlayer', ((playerid, clickedplayerid, source) => func(SampPlayers_1.SampPlayers.getClass(playerid), SampPlayers_1.SampPlayers.getClass(clickedplayerid), source)));
};
exports.OnPlayerClickPlayer = OnPlayerClickPlayer;
const OnPlayerSelectObject = (func) => {
    samp.on('OnPlayerSelectObject', ((playerid, type, objectid, modelid, fX, fY, fZ) => func(SampPlayers_1.SampPlayers.getClass(playerid), type, objectid, modelid, fX, fY, fZ)));
};
exports.OnPlayerSelectObject = OnPlayerSelectObject;
const OnPlayerWeaponShot = (func) => {
    samp.on('OnPlayerWeaponShot', ((playerid, weaponid, hittype, hitid, fX, fY, fZ) => func(SampPlayers_1.SampPlayers.getClass(playerid), weaponid, hittype, hitid, fX, fY, fZ)));
};
exports.OnPlayerWeaponShot = OnPlayerWeaponShot;
const OnClientCheckResponse = (func) => {
    samp.on('OnClientCheckResponse', ((playerid, actionid, memaddr, retndata) => func(SampPlayers_1.SampPlayers.getClass(playerid), actionid, memaddr, retndata)));
};
exports.OnClientCheckResponse = OnClientCheckResponse;
const OnScriptCash = (func) => {
    samp.on('OnScriptCash', ((playerid, amount, source) => func(SampPlayers_1.SampPlayers.getClass(playerid), amount, source)));
};
exports.OnScriptCash = OnScriptCash;
